components/gcp-click-to-deploy/src/DeployForm.tsx

```diff
@@ -1076,28 +1091,31 @@ export default class DeployForm extends React.Component<any, DeployFormState> {
             {
                 name: 'accessToken',
                 secretSource: {literalSource: {value: saToken}}
+            },
+            {
+                name: KUBEFLOW_OAUTH,
+                secretSource: {literalSource: {value: clientSecret || ''}}
             }
         ];
         const gcpPlugin = {
             name: 'gcp',
             spec: {
                 auth: {},
-                createPipelinePersistentStorage: permanentStorage,
-                username: saClientId,
+                createPipelinePersistentStorage: permanentStorage
             }
         };
         if (ingress === IngressType.BasicAuth) {
             gcpPlugin.spec.auth = {
                 basicAuth: {
-                    password: this.state.password,
+                    password: btoa(encryptPassword(this.state.password)),
```

Comment:
password field actually has type
```
&kfdefsv2.SecretRef
```
@jlewi 
What's your suggestion about passing password hash instead of raw password?

Reply:
In the latest commit, I switched this to use a SecretRef.